tests/conftest.py

```diff
@@ -174,6 +174,43 @@ def a10g_gpu_cluster():
     ).up_if_not()
 
 
+@pytest.fixture(scope="session")
+def password_cluster():
+    sky_cluster = rh.cluster("temp-rh-password", instance_type="CPU:4").save()
+    if not sky_cluster.is_up():
+        sky_cluster.up()
+
+        # set up password on remote
+        sky_cluster.run(
+            [
+                [
+                    'sudo sed -i "/^[^#]*PasswordAuthentication[[:space:]]no/c\PasswordAuthentication yes" '
+                    "/etc/ssh/sshd_config"
+                ]
+            ]
+        )
+        sky_cluster.run(["sudo /etc/init.d/ssh force-reload"])
+        sky_cluster.run(["sudo /etc/init.d/ssh restart"])
+        sky_cluster.run(
+            ["(echo 'cluster-pass' && echo 'cluster-pass') | sudo passwd ubuntu"]
+        )
+        sky_cluster.run(["pip uninstall skypilot runhouse -y", "pip install pytest"])
+        sky_cluster.run(["rm -rf runhouse/"])
+
+    # instantiate byo cluster with password
+    ssh_creds = {"ssh_user": "ubuntu", "password": "cluster-pass"}
+    cluster = rh.cluster(
+        name="rh-password", ips=[sky_cluster.address], ssh_creds=ssh_creds
+    ).save()
+
+    return cluster
+
+
+parametrize_cpu_clusters = pytest.mark.parametrize(
+    "cluster", ["cpu_cluster", "password_cluster"], indirect=True
+)
```

Comment:
Also, it might be clearer to leave the name of the parameterized fixture as "cpu_cluster" and chance the name of the existing cpu_cluster to "ondemand_cpu_cluster" ? If that's an even more dramatic change than what you already have here then don't bother.

Reply:
sure, I went ahead and updated the var names. the easiest way to toggle currently would be adding `-k "not [password_cluster]"` to your pytest command, since all password_cluster tests are part of a parametrize currently.